03-phenology/notebooks/01-download-Daymet4.ipynb

```diff
@@ -0,0 +1,730 @@
+   "source": [
+    "# save catalog to storage\n",
+    "catalog.normalize_and_save(\n",
+    "    f\"{dcache_root}/{catalog_id}\",\n",
+    "    catalog_type=pystac.CatalogType.SELF_CONTAINED,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be4e2c7b-1b2b-4534-adce-9715d05e75f4",
+   "metadata": {},
+   "source": [
+    "## 3. Retrieve data for spring index calculation\n",
+    "\n",
+    "### 3.1 Overview \n",
+    "\n",
+    "After having created a STAC catalog with the metadata for the  Daymet dataset, we now procede to retrieve the files that are relevant for the extended Spring Index model calculations over our area of interest (i.e. the continental North America). Only for this region, we thus retrieve the assets corresponding to the following three parameters: maximum temperature, minimum temperature, and duration of the daylight period. \n",
+    "\n",
+    "The data files are stored beside the metadata files on the dCache storage. Links within the catalog are updated to point to the files that have been retrieved to SURF infrastructures."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65707e5b-2704-4896-8ad5-a45571af6876",
+   "metadata": {},
+   "source": [
+    "### 3.2 Dowload\n",
+    "\n",
+    "We first select the only sub-catalog with the items related to the continental North America (collection ID: `region-na`): "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "72fc2707-b43b-416f-8c28-218879d03881",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_na = catalog.get_child(\"region-na\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b61422ca-9b13-45c4-84c4-f23fe517652b",
+   "metadata": {},
+   "source": [
+    "Finally, we download the assets to the storage and save the catalog with the updated links:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "64d34f99-10d0-420d-a3e0-ae137018d232",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for asset_key in (\"tmin\", \"tmax\", \"dayl\"):\n",
+    "    copy_asset(\n",
+    "        catalog=collection_na,\n",
+    "        asset_key=asset_key,\n",
+    "        update_catalog=True,\n",
+    "        max_workers=4\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4b4381cd-24bc-4e39-951e-72561682f5c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "catalog.normalize_and_save(f\"{dcache_root}/{catalog_id}\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.9.13"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
Is it interesting here to add some time scales for how long the download takes in case someone wants to use this for their own computations?

Reply:
Good point - added!